File: backend/app/schemas/supervision.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class SupervisionCheckRequest(BaseModel):
    image_base64: str
    session_id: str


class SupervisionCheckResponse(BaseModel):
    status: str
    confidence: float
    face_count: int
    overlay_base64: Optional[str] = None


class SupervisionSessionCreate(BaseModel):
    session_id: str
    started_at: datetime


class SupervisionSessionStats(BaseModel):
    session_id: str
    total_checks: int
    focused_count: int
    distracted_count: int
    absent_count: int
    unknown_count: int
    focus_rate: float
    duration_seconds: int


class SupervisionDailyStats(BaseModel):
    date: str
    total_focused_seconds: int
    total_checks: int
    focus_rate: float

File: frontend/src/api/supervision.ts
import axios from './axios'

export interface SupervisionCheckRequest {
  image_base64: string
  session_id: string
}

export interface SupervisionCheckResponse {
  status: string
  confidence: number
  face_count: number
  overlay_base64?: string | null
}

export interface SupervisionSessionStart {
  session_id: string
  started_at: string
}

export interface SupervisionSessionStats {
  session_id: string
  total_checks: number
  focused_count: number
  distracted_count: number
  absent_count: number
  unknown_count: number
  focus_rate: number
  duration_seconds: number
}

export interface SupervisionDailyStats {
  date: string
  total_focused_seconds: number
  total_checks: number
  focus_rate: number
}

export async function startSupervisionSession(): Promise<SupervisionSessionStart> {
  return await axios.post('/supervision/session/start')
}

export async function checkSupervision(data: SupervisionCheckRequest): Promise<SupervisionCheckResponse> {
  return await axios.post('/supervision/check', data)
}

export async function getSupervisionSessionStats(sessionId: string): Promise<SupervisionSessionStats> {
  return await axios.get(`/supervision/session/${sessionId}/stats`)
}

export async function getSupervisionDailyStats(date?: string): Promise<SupervisionDailyStats> {
  return await axios.get('/supervision/daily-stats', { params: date ? { target_date: date } : {} })
}
